server/src/main/java/org/elasticsearch/index/IndexSettings.java

```diff
@@ -702,7 +702,7 @@ public IndexSettings(final IndexMetadata indexMetadata, final Settings nodeSetti
         this.indexMetadata = indexMetadata;
         numberOfShards = settings.getAsInt(IndexMetadata.SETTING_NUMBER_OF_SHARDS, null);
         mode = isTimeSeriesModeEnabled() ? scopedSettings.get(MODE) : IndexMode.STANDARD;
-        this.timestampBounds = TIME_SERIES_START_TIME.exists(settings) ? new TimestampBounds(scopedSettings) : null;
+        this.timestampBounds = mode == IndexMode.TIME_SERIES ? new TimestampBounds(scopedSettings) : null;
```

Comment:
Do you want to delegate this to a method on `IndexMode`? Less hard tests is nice.